html_css/v2/foundations/css-foundations.md

```diff
@@ -0,0 +1,425 @@
+
+Then inside of the newly created `styles.css` file, you'll notice the syntax is basically the same as the internal CSS method. The difference here being we no longer need those `<style>` tags (in fact, there should be no HTML tags within an external CSS file). `styles.css` is just what we went with as the file name here. You can name the file whatever you want as long as the file type is `.css`, though "style" or "styles" is most commonly used.
+
+Why is this the preferred method?
+
+1. It keeps our HTML and CSS separated, which results in the HTML file being smaller and makes things look cleaner
+2. We only need to edit the CSS in *one* place, which is especially handy for websites with multiple pages that all have similar styles.
+
+### Selectors
+
+Selectors simply refer to the HTML elements in which CSS rules apply; they're what is actually being "selected" for each rule. The following sub-sections don't cover every selector available, but they're definitely ones you should get comfortable using first.
+
+#### Universal Selector
+
+The universal selector will select elements of any type, hence the name "universal", and the syntax for it is a simple asterisk. In the example below, every element would have the `color: purple;` style applied to it.
+
+~~~css
+* {
+  color: purple;
+}
+~~~
+
+#### Type Selectors
+
+A type selector (or element selector) you've already seen, as each code snippet up to this point has used a type selector. This selector will select all elements of the given element type, and the syntax is just the name of the element. Just to make sure it's clear, here's another example:
+
+~~~html
+<!-- index.html -->
+
+<div>Hello, World!</div>
+<div>Hello again!</div>
+<p>Hi...</p>
+<div>Okay, bye.</div>
+~~~
+~~~css
+/* styles.css */
+
+div {
+  color: white;
+}
+~~~
+
+Here, all three `<div>` elements would be selected, while the `<p>` element wouldn't be.
+
+#### Class Selectors
+
+Class selectors will select all elements with the given class, which is just an attribute you place on an HTML element. Here's how you would add a class in an HTML file and select it in CSS:
+
+~~~html
+<!-- index.html -->
+
+<div class="alert-text">
+  Please agree to our terms of service.
+</div>
+~~~
+~~~css
+/* styles.css */
+
+.alert-text {
+  color: red;
+}
+~~~
+
+Note the syntax for class selectors: a period immediately followed by the case-sensitive value of the class attribute. Classes aren't required to be unique, so you can use the same class on as many elements as you want.
+
+Another thing you may have noticed is the kebab-case style. We separate a multi-worded class with hyphens because whitespace is reserved for separating a list of several classes on a single element, such as `class="alert-text severe-alert"`. Kebab-case isn't exactly required, but it tends to be the convention that developers opt for.
```

Comment:
I think i'd leave out the kebab-case mention here.. this is the very first coding our students have done so they probably aren't ready for kebabs or snakes or camels or whatever lol.  I'd just say something like "don't put spaces in there, use hyphens" and leave it at that for the moment.